log/tmfmt_logger.go

```diff
@@ -0,0 +1,116 @@
+package log
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/go-logfmt/logfmt"
+)
+
+type tmfmtEncoder struct {
+	*logfmt.Encoder
+	buf bytes.Buffer
+}
+
+func (l *tmfmtEncoder) Reset() {
+	l.Encoder.Reset()
+	l.buf.Reset()
+}
+
+var tmfmtEncoderPool = sync.Pool{
+	New: func() interface{} {
+		var enc tmfmtEncoder
+		enc.Encoder = logfmt.NewEncoder(&enc.buf)
+		return &enc
+	},
+}
+
+type tmfmtLogger struct {
+	w io.Writer
+}
+
+// NewTmFmtLogger returns a logger that encodes keyvals to the Writer in
```

Comment:
actually the standard is to make uppercase abbreviations. so probably should be NewTMFmtLogger